From 845f14c57fd50ec7e94f59bb1b5ad0cd1ecdbdd1 Mon Sep 17 00:00:00 2001
From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Tue, 30 Aug 2016 15:47:05 -0400
Subject: [PATCH] rtl8xxxu: Early enable of WEP/TKIP security on 8188eu

This matches action taken in the vendor driver, however it is unclear
why this is done.

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---
 drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c
+++ b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c
@@ -1394,6 +1394,9 @@ static void rtl8188e_usb_quirks(struct r
 	val32 = rtl8xxxu_read32(priv, REG_TXDMA_OFFSET_CHK);
 	val32 |= TXDMA_OFFSET_DROP_DATA_EN;
 	rtl8xxxu_write32(priv, REG_TXDMA_OFFSET_CHK, val32);
+
+	/* Pre-TX enable WEP/TKIP security */
+	rtl8xxxu_write8(priv, REG_EARLY_MODE_CONTROL_8188E + 3, 0x01);
 }
 
 struct rtl8xxxu_fileops rtl8188eu_fops = {
